src/averageAges.js

```diff
@@ -15,11 +15,15 @@
  * @return {number}
  */
 function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
+  const men = century ? people.filter(person => person.sex === 'm'
+  && Math.ceil(person.died / 100) === century)
+    : people.filter(person => person.sex === 'm');
+
+  const age = men.map(date => date.died - date.born);
+  const mean = ((age.reduce((num1, num2) =>
+    num1 + num2)) / age.length).toFixed(2);
```

Comment:
mean? mb average age?
```suggestion
  const mean = ((age.reduce((sum, value) =>
    sum + value)) / age.length).toFixed(2);
```